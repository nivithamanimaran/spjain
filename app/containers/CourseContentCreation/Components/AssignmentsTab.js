import React from 'react';
import { Box, Grid } from '@material-ui/core';
import AssignmentCard from './AssignmentsCard';
import EmptyDataComponent from './EmptyDataComponent';

export default function AssignmentsTab(props) {
  const { data } = props;
  return (
    <Box>
      <Grid container spacing={2}>
        {Array.isArray(data) && data.length > 0 ? (
          data.map(obj => (
            <Grid item>
              <AssignmentCard
                name={obj?.name}
                startDate={obj?.allowsubmissionsfromdate}
                dueDate={obj?.duedate}
                url={obj?.url}
              />
            </Grid>
          ))
        ) : (
          <EmptyDataComponent />
        )}
      </Grid>
    </Box>
  );
}
