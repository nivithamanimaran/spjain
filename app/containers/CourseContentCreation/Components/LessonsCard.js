import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import Typography from '@material-ui/core/Typography';
import { Box, Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import CardActionArea from '@material-ui/core/CardActionArea';

const useStyles = makeStyles(theme => ({
  root: {
    zIndex: -1,
    display: 'flex',
    flexDirection: 'row',
    width: '100%',
    maxWidth: '23vw',
    flexWrap: 'wrap',
    // borderLeft: `3px solid ${theme.palette.KenColors.primary}`,
    backgroundColor: theme.palette.KenColors.tertiaryBlue40,
    border: `1px solid ${theme.palette.KenColors.neutral20}`,
    margin: 8,
    [theme.breakpoints.only('xs')]: {
      flexWrap: 'none',
      maxWidth: 'none',
      margin: '0px 0px 8px 0px',
      padding: 0,
    },
  },
  title: {
    color: theme.palette.KenColors.primary,
    fontSize: '14px',
    lineHeight: '22px',
    textTransform: 'capitalize',
  },
  date: {
    color: theme.palette.KenColors.neutral900,
    fontSize: '10px',
    lineHeight: '14px',
    marginTop: 16,
  },
  facultyTasks: {
    width: '100%',
    // minWidth: '350px',
  },
}));

export default function LessonsCrad(props) {
  const { name, dueDate, startDate, url } = props;
  const classes = useStyles();
  const { t } = useTranslation();

  const goToContent = url => {
    window.open(url, '_blank');
  };

  //   const getTitle = title => {
  //     if (typeof title !== 'string') return null;

  //     if (title.length <= 30) {
  //       return title;
  //     } else {
  //       return title.substring(0, 28).trim() + '...';
  //     }
  //   };

  return (
    <Card
      elevation={0}
      className={classes.root}
      onClick={() => goToContent(url)}
    >
      <CardActionArea>
        <Grid container direction="column">
          <Box p={2} className={classes.facultyTasks}>
            <Grid item>
              <Typography
                className={classes.title}
                component="span"
                data-testid="lesson-title"
              >
                {/* {getTitle(data?.name)} */}
                {name}
              </Typography>
            </Grid>
            <Grid container item alignItems="center" spacing={1}>
              <Grid item xs={12} sm={12} md={6}>
                <Typography
                  className={classes.date}
                  data-testid="lesson-start-date"
                >
                  <span>{t('Starts_On')}:</span> {startDate}{' '}
                </Typography>
              </Grid>
              <Grid item item xs={12} sm={12} md={6}>
                <Typography
                  className={classes.date}
                  data-testid="lesson-due-date"
                >
                  <span>{t('Deadline')}:</span> {dueDate}{' '}
                </Typography>
              </Grid>
            </Grid>
          </Box>
        </Grid>
      </CardActionArea>
    </Card>
  );
}
