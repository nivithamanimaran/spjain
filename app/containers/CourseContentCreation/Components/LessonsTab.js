import React from 'react';
import { Box, Grid } from '@material-ui/core';
import LessonsCard from './LessonsCard';
import EmptyDataComponent from './EmptyDataComponent';

export default function LessonsTab(props) {
  const { data } = props;
  return (
    <Box>
      <Grid container spacing={2}>
        {Array.isArray(data) && data.length > 0 ? (
          data.map(obj => (
            <Grid item>
              <LessonsCard
                name={obj?.name}
                startDate={obj?.available}
                dueDate={obj?.deadline}
                url={obj?.url}
              />
            </Grid>
          ))
        ) : (
          <EmptyDataComponent />
        )}
      </Grid>
    </Box>
  );
}
