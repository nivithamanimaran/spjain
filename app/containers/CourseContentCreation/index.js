import React, { useEffect, useState } from 'react';
import { Grid, Box } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import KenClassList from '../../global_components/KenClassList';
import { getUserDetails } from '../../utils/helpers/storageHelper';
import { facultyActivities, StudentActivities } from '../../utils/ApiService';
import { KEY_STATUS, KEY_USER_TYPE } from '../../utils/constants';
import KenTabs from '../../components/KenTabs';
import AssignmentsTab from './Components/AssignmentsTab';
import AssessmentsTab from './Components/AssessmentsTab';
import LessonsTab from './Components/LessonsTab';
import AssignmentIcon from '@material-ui/icons/Assignment';
import AssessmentIcon from '@material-ui/icons/Assessment';
import MenuBookIcon from '@material-ui/icons/MenuBook';
import KenLoader from '../../components/KenLoader';
import { useTranslation } from 'react-i18next';
import KenButton from '../../global_components/KenButton';

const useStyles = makeStyles(theme => ({
  root: {
    zIndex: -1,
  },
  dropDown: {
    display: 'flex',
  },
  button: {
    fontWeight: '600',
    fontSize: '14px',
    lineHeight: '20px',
  },
  icons: {
    fontSize: '18px',
    marginRight: '10px',
  },
  title: {
    display: 'flex',
    alignItems: 'center',
  },
}));

export default function ContentCreationPage() {
  const classes = useStyles();
  const [selectedSection, setSelectedSection] = useState();
  const [activityUrls, setActivityUrls] = useState();
  const [tabData, setTabData] = useState();
  const [loading, setLoading] = useState(false);
  const { t } = useTranslation();

  const getSelectedSection = section => {
    setSelectedSection(section);
  };

  const getTabData = (assignsArray, assesArray, lessonArray, notesArray) => {
    return [
      {
        title: (
          <Box className={classes.title}>
            <MenuBookIcon className={classes.icons} />
            <span>{`Lessons (${lessonArray?.length || '0'})`}</span>
          </Box>
        ),
        content: <LessonsTab data={lessonArray} />,
      },
      {
        title: (
          <Box className={classes.title}>
            <AssignmentIcon className={classes.icons} />
            <span>{`Assignments (${assignsArray?.length || '0'})`}</span>
          </Box>
        ),
        content: <AssignmentsTab data={assignsArray} />,
      },
      {
        title: (
          <Box className={classes.title}>
            <AssessmentIcon className={classes.icons} />
            <span>{`Assessments (${assesArray?.length || '0'})`}</span>
          </Box>
        ),
        content: <AssessmentsTab data={assesArray} />,
      },
    ];
  };

  const goToContent = () => {
    let urlObj = activityUrls.find(url => url.id === selectedSection.Id);
    if (urlObj) {
      window.open(urlObj.url, '_blank');
    }
  };

  useEffect(() => {
    setLoading(true);
    const user = getUserDetails();
    const contactID = user?.ContactId;
    const connectionId = selectedSection?.Id;
    if (connectionId && contactID) {
      if (
        user?.Type?.toLocaleLowerCase() ===
        KEY_USER_TYPE.faculty?.toLocaleLowerCase()
      ) {
        facultyActivities(contactID, connectionId)
          .then(res => {
            setLoading(false);
            const activityLinks = [];
            let assignmentsArray = [];
            let assessmentsArray = [];
            let lessonsArray = [];
            if (res.status !== KEY_STATUS.failed) {
              res?.courses.forEach(el => {
                let subject = el.fullname;
                if (el.activities.assignment) {
                  el.activities.assignment.forEach(ele => {
                    assignmentsArray.push({ ...ele, subject });
                  });
                }
                if (el.activities.quizzes) {
                  el.activities.quizzes.forEach(ele => {
                    assessmentsArray.push({ ...ele, subject });
                  });
                }
                if (el.activities.lesson) {
                  el.activities.lesson.forEach(ele => {
                    lessonsArray.push({ ...ele, subject });
                  });
                }
                if (el?.url) {
                  activityLinks.push({
                    subject: el.fullname,
                    url: el.url,
                    id: el.connectionid,
                  });
                }
              });
            }
            setTabData(
              getTabData(assignmentsArray, assessmentsArray, lessonsArray)
            );
            setActivityUrls(activityLinks);
          })
          .catch(err => {
            console.log('error in facultyActivities', err);
            setActivityUrls([]);
            setLoading(false);
          });
      } else if (
        user?.Type?.toLocaleLowerCase() ===
        KEY_USER_TYPE.student?.toLocaleLowerCase()
      ) {
        StudentActivities(contactID, connectionId)
          .then(res => {
            setLoading(false);
            const activityLinks = [];
            let assignmentsArray = [];
            let assessmentsArray = [];
            let lessonsArray = [];
            if (res.status !== KEY_STATUS.failed) {
              res?.courses.forEach(el => {
                let subject = el.fullname;
                if (el.activities.assignment) {
                  el.activities.assignment.forEach(ele => {
                    assignmentsArray.push({ ...ele, subject });
                  });
                }
                if (el.activities.quizzes) {
                  el.activities.quizzes.forEach(ele => {
                    assessmentsArray.push({ ...ele, subject });
                  });
                }
                if (el.activities.lesson) {
                  el.activities.lesson.forEach(ele => {
                    lessonsArray.push({ ...ele, subject });
                  });
                }
                if (el?.url) {
                  activityLinks.push({
                    subject: el.fullname,
                    url: el.url,
                    id: el.connectionid,
                  });
                }
              });
            }
            setTabData(
              getTabData(assignmentsArray, assessmentsArray, lessonsArray)
            );
            setActivityUrls(activityLinks);
          })
          .catch(err => {
            console.log('error in facultyActivities', err);
            setActivityUrls([]);
            setLoading(false);
          });
      }
    }
  }, [selectedSection]);

  return (
    <>
      {loading && <KenLoader />}
      <Grid container direction="column" className={classes.root} spacing={2}>
        <Grid item>
          <Box>
            <KenClassList
              getSelectedSection={getSelectedSection}
              title={t('headings:Course_Content')}
              listTitle={t('headings:Your_Subjects')}
            />
          </Box>
        </Grid>
        {getUserDetails()?.Type?.toLowerCase() ===
        KEY_USER_TYPE.faculty.toLocaleLowerCase() ? (
          <Grid item style={{ textAlign: 'right' }}>
            <Box>
              <KenButton
                label={t('labels:Add_Content')}
                onClick={() => goToContent()}
                variant="primary"
                style={{ marginBottom: '8px' }}
              />
            </Box>
          </Grid>
        ) : null}
        {/* <Grid item>
        
        </Grid> */}
      </Grid>
      {tabData && (
        <Box mt={2}>
          <KenTabs
            data={tabData}
            tabPanelProps={{
              style: { minHeight: '250px', position: 'relative' },
            }}
          />
        </Box>
      )}
    </>
  );
}
