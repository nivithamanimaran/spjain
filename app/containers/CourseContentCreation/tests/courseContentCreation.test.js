import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import ThemeProvider from '../../../utils/themeProvider';

import CourseContentCreation from '../index';
import AssessmentsTab from '../Components/AssessmentsTab';
import LessonsTab from '../Components/LessonsTab';
import AssignmentsTab from '../Components/AssignmentsTab';
import AssignmentsCard from '../Components/AssignmentsCard';
import LessonsCard from '../Components/LessonsCard';

describe('<CourseContentCreation />', () => {
  it('CourseContentCreation should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ThemeProvider>
          <CourseContentCreation />
        </ThemeProvider>
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});

describe('<AssessmentsTab />', () => {
  it('AssessmentsTab should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ThemeProvider>
          <AssessmentsTab />
        </ThemeProvider>
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});

describe('<AssignmentsTab />', () => {
  it('AssignmentsTab should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ThemeProvider>
          <AssignmentsTab />
        </ThemeProvider>
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});

describe('<LessonsTab />', () => {
  it('LessonsTab should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ThemeProvider>
          <LessonsTab />
        </ThemeProvider>
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});

describe('<AssignmentsCard />', () => {
  it('AssignmentsCard should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ThemeProvider>
          <AssignmentsCard />
        </ThemeProvider>
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});

describe('<LessonsCard />', () => {
  it('LessonsCard should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ThemeProvider>
          <LessonsCard />
        </ThemeProvider>
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});

const data = [
  {
    name: 'Math',
    duedate: '12/07/2021',
    allowsubmissionsfromdate: '12/05/2021',
  },
];

it('AssessmentsTab renders correctly with props', () => {
  const {
    container: { firstChild },
  } = render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <AssessmentsTab data={data} />
      </ThemeProvider>
    </IntlProvider>
  );

  expect(firstChild).toMatchSnapshot();
});

it('AssignmentsTab renders correctly with props', () => {
  const {
    container: { firstChild },
  } = render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <AssignmentsTab data={data} />
      </ThemeProvider>
    </IntlProvider>
  );

  expect(firstChild).toMatchSnapshot();
});

it('LessonsTab renders correctly with props', () => {
  const {
    container: { firstChild },
  } = render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <LessonsTab data={data} />
      </ThemeProvider>
    </IntlProvider>
  );

  expect(firstChild).toMatchSnapshot();
});

it('AssignmentsCard renders correctly with props', () => {
  render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <AssignmentsCard
          name="Hello"
          startDate="12/07/2021"
          dueDate="12/08/2021"
        />
      </ThemeProvider>
    </IntlProvider>
  );
  const { queryByTestId } = render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <AssignmentsCard
          name="Hello"
          startDate="12/07/2021"
          dueDate="12/08/2021"
        />
      </ThemeProvider>
    </IntlProvider>
  );

  const assignmentTitle = queryByTestId('assignment-title');
  expect(assignmentTitle).toBeTruthy();
  expect(assignmentTitle.textContent).toBe('Hello');

  const assignmentStartDate = queryByTestId('assignment-start-sate');
  expect(assignmentStartDate).toBeTruthy();
  expect(assignmentStartDate.textContent).toBe('Starts_On: 12/07/2021 ');

  const assignmentDueDate = queryByTestId('assignment-due-date');
  expect(assignmentDueDate).toBeTruthy();
  expect(assignmentDueDate.textContent).toBe('Due_On: 12/08/2021 ');
});

it('LessonsCard renders correctly with props', () => {
  render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <LessonsCard name="Hello" startDate="12/07/2021" dueDate="12/08/2021" />
      </ThemeProvider>
    </IntlProvider>
  );
  const { queryByTestId } = render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <LessonsCard name="Hello" startDate="12/07/2021" dueDate="12/08/2021" />
      </ThemeProvider>
    </IntlProvider>
  );

  const lessonTitle = queryByTestId('lesson-title');
  expect(lessonTitle).toBeTruthy();
  expect(lessonTitle.textContent).toBe('Hello');

  const lessonStartDate = queryByTestId('lesson-start-date');
  expect(lessonStartDate).toBeTruthy();
  expect(lessonStartDate.textContent).toBe('Starts_On: 12/07/2021 ');

  const lessonDueDate = queryByTestId('lesson-due-date');
  expect(lessonDueDate).toBeTruthy();
  expect(lessonDueDate.textContent).toBe('Deadline: 12/08/2021 ');
});
