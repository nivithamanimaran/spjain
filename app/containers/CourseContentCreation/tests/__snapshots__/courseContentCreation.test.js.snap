// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AssessmentsTab /> AssessmentsTab should render and match the snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-23"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiBox-root MuiBox-root-24"
      style="width: 100%; position: absolute; top: 40%;"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="text-align: center;"
      >
        labels:No_Data
      </p>
    </div>
  </div>
</div>
`;

exports[`<AssignmentsCard /> AssignmentsCard should render and match the snapshot 1`] = `
<div
  class="MuiPaper-root MuiCard-root makeStyles-root-29 MuiPaper-elevation0 MuiPaper-rounded"
>
  <button
    class="MuiButtonBase-root MuiCardActionArea-root"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        class="MuiBox-root MuiBox-root-33 makeStyles-facultyTasks-32"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <span
            class="MuiTypography-root makeStyles-title-30 MuiTypography-body1"
            data-testid="assignment-title"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        />
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root makeStyles-date-31 MuiTypography-body1"
              data-testid="assignment-start-sate"
            >
              <span>
                Starts_On
                :
              </span>
               
               
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root makeStyles-date-31 MuiTypography-body1"
              data-testid="assignment-due-date"
            >
              <span>
                Due_On
                :
              </span>
               
               
            </p>
          </div>
        </div>
      </div>
    </div>
    <span
      class="MuiCardActionArea-focusHighlight"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`<AssignmentsTab /> AssignmentsTab should render and match the snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-25"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiBox-root MuiBox-root-26"
      style="width: 100%; position: absolute; top: 40%;"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="text-align: center;"
      >
        labels:No_Data
      </p>
    </div>
  </div>
</div>
`;

exports[`<CourseContentCreation /> CourseContentCreation should render and match the snapshot 1`] = `
<div
  class="makeStyles-root-22"
>
  <div
    class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</div>
`;

exports[`<LessonsCard /> LessonsCard should render and match the snapshot 1`] = `
<div
  class="MuiPaper-root MuiCard-root makeStyles-root-34 MuiPaper-elevation0 MuiPaper-rounded"
>
  <button
    class="MuiButtonBase-root MuiCardActionArea-root"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        class="MuiBox-root MuiBox-root-38 makeStyles-facultyTasks-37"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <span
            class="MuiTypography-root makeStyles-title-35 MuiTypography-body1"
            data-testid="lesson-title"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        />
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root makeStyles-date-36 MuiTypography-body1"
              data-testid="lesson-start-date"
            >
              <span>
                Starts_On
                :
              </span>
               
               
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root makeStyles-date-36 MuiTypography-body1"
              data-testid="lesson-due-date"
            >
              <span>
                Deadline
                :
              </span>
               
               
            </p>
          </div>
        </div>
      </div>
    </div>
    <span
      class="MuiCardActionArea-focusHighlight"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`<LessonsTab /> LessonsTab should render and match the snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-27"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiBox-root MuiBox-root-28"
      style="width: 100%; position: absolute; top: 40%;"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="text-align: center;"
      >
        labels:No_Data
      </p>
    </div>
  </div>
</div>
`;

exports[`AssessmentsTab renders correctly with props 1`] = `
<div
  class="MuiBox-root MuiBox-root-39"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-40 MuiPaper-elevation0 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
          >
            <div
              class="MuiBox-root MuiBox-root-44 makeStyles-facultyTasks-43"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <span
                  class="MuiTypography-root makeStyles-title-41 MuiTypography-body1"
                  data-testid="assignment-title"
                >
                  Math
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              />
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <p
                    class="MuiTypography-root makeStyles-date-42 MuiTypography-body1"
                    data-testid="assignment-start-sate"
                  >
                    <span>
                      Starts_On
                      :
                    </span>
                     
                    12/05/2021
                     
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <p
                    class="MuiTypography-root makeStyles-date-42 MuiTypography-body1"
                    data-testid="assignment-due-date"
                  >
                    <span>
                      Due_On
                      :
                    </span>
                     
                    12/07/2021
                     
                  </p>
                </div>
              </div>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`AssignmentsTab renders correctly with props 1`] = `
<div
  class="MuiBox-root MuiBox-root-45"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-46 MuiPaper-elevation0 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
          >
            <div
              class="MuiBox-root MuiBox-root-50 makeStyles-facultyTasks-49"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <span
                  class="MuiTypography-root makeStyles-title-47 MuiTypography-body1"
                  data-testid="assignment-title"
                >
                  Math
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              />
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <p
                    class="MuiTypography-root makeStyles-date-48 MuiTypography-body1"
                    data-testid="assignment-start-sate"
                  >
                    <span>
                      Starts_On
                      :
                    </span>
                     
                    12/05/2021
                     
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <p
                    class="MuiTypography-root makeStyles-date-48 MuiTypography-body1"
                    data-testid="assignment-due-date"
                  >
                    <span>
                      Due_On
                      :
                    </span>
                     
                    12/07/2021
                     
                  </p>
                </div>
              </div>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`LessonsTab renders correctly with props 1`] = `
<div
  class="MuiBox-root MuiBox-root-51"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-52 MuiPaper-elevation0 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
          >
            <div
              class="MuiBox-root MuiBox-root-56 makeStyles-facultyTasks-55"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <span
                  class="MuiTypography-root makeStyles-title-53 MuiTypography-body1"
                  data-testid="lesson-title"
                >
                  Math
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              />
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <p
                    class="MuiTypography-root makeStyles-date-54 MuiTypography-body1"
                    data-testid="lesson-start-date"
                  >
                    <span>
                      Starts_On
                      :
                    </span>
                     
                     
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <p
                    class="MuiTypography-root makeStyles-date-54 MuiTypography-body1"
                    data-testid="lesson-due-date"
                  >
                    <span>
                      Deadline
                      :
                    </span>
                     
                     
                  </p>
                </div>
              </div>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
